### DISCUSSION TITLE
Allegro+LAMMPS Distributed

### DISCUSSION THREAD

**Discussion by npiroozan** [mir-group/allegro#39] [Q&A] [answered]
Hello,

I hope you are well.

Upon building the entire framework:

PyTorch
Nequip
Allegro
Pair_Allegro
LAMMPS

On CPU only, is there any issue in your view to running the LAMMPS+Allegro simulation in distributed across multiple nodes?

Thank you!

**Comment by npiroozan** [comment, marked as answer]
Fixed the Issue

**Reply by Linux-cpp-lisp** [reply]
Glad to hear, and yes there is no issue running across multiple CPU ranks, although please not that CPU performance is not currently a target for us. 

Linking also for reference: https://github.com/mir-group/nequip/discussions/303

**Reply by npiroozan** [reply]
I understand, thank you for the reply. I can work on cpu improvement.

What areas of the code would you advise to focus on?

**Reply by Linux-cpp-lisp** [reply]
It is possible that using PyTorch 2.0 with the _new_ Intel Extension for PyTorch might get you somewhere.  I'll also be sure to post again if we have relevant efforts in this direction.

Otherwise, having built PyTorch from scratch against MKL, I would spend some time playing with the balance between LAMMPS ranks and OpenMP/MKL threads to see what gives optimal performance, maybe?

**Reply by npiroozan** [reply]
Agreed, perfect, thank you. Building PyTorch from source and using ipex helped quite a bit. Will focus on tuning the LAMMPS settings relating to OMP, etc.

Will be in touch.

**Reply by Linux-cpp-lisp** [reply]
Sounds good--- @anjohan mentioned that he thought the extreme of one rank per core and one OpenMP thread _might_ be best, so might be worth trying that as well.  No harm probably in trying the other extreme either.

**Reply by npiroozan** [reply]
Excellent. Yeah, Anders seems to be correct actually. Scaling across many cpu nodes would require a different approach of course. I doubt we could do 1 rank per core across many CPUs and maintain performance.
